### PR TITLE
Feature/fdp cache persistence

### PR DESCRIPTION
**Pull request by kburger**
Functionality as intended in DEM9, "As a user, I want to have FDPs that are cached in the demonstrator, persisted...". The data is taken from blazegraph, and a simple query is done to list all known FDPs. I have had some issued running it properly locally, so if you see the chance please fork this locally and run it before merging this.

This PR will have to be updated after PR #8 and PR #9 are accepted and merged.

Also, I'm try to see if I can merge from my feature branch directly into the origin develop branch. While creating the PR, github warns it cannot be merged automatically.

### CODE REVIEW

**Comment by kburger** [comment]
Added an additional commit with a (silly) bugfix, I forgot to add back the `$scope.fairDataPoints` initialization.

File: demo/cache/cache-controller.js [hunk 1]
```diff
@@ -1,8 +1,35 @@
-app.controller('CacheCtrl', function($scope, $rootScope, $http, FDP, Caching, Log) {    
+app.controller('CacheCtrl', function($scope, $rootScope, $http, FDP, Caching, Log, HttpEndpoint, File) {    
   $scope.isCachingStarted = false;
-  $scope.fairDataPoints=[{name: "RD connect FDP", url:"http://localhost:8084/fairdatapoint/fdp"}];
+  $scope.fairDataPoints = [];
   $scope.searchEngineUrl=null;
```
Comment (rajaram5) [inline demo/cache/cache-controller.js]: Some genius wrote this line of code :-D, could you be kind enough to add space between the "="?, BTW, that genius is me ;-)

File: sparqlQueries/listCachedFdps.sparql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+PREFIX r3d: <http://www.re3data.org/schema/3-0#>
+PREFIX dct: <http://purl.org/dc/terms/>
+
+SELECT DISTINCT ?fdp ?name {
+    ?fdp r3d:dataCatalog ?catalog ;
```
Comment (rajaram5) [inline sparqlQueries/listCachedFdps.sparql]: Maybe you can get the type of fdpURI right? This statement returns empty fdp URI if the repository doesn't have any catalogies.

**Review by rajaram5 — CHANGES_REQUESTED** [review]

**Review by nunogit — APPROVED** [review]

File: sparqlQueries/listCachedFdps.sparql (hunk 2 above)
Comment (kburger) [reply sparqlQueries/listCachedFdps.sparql]: Good point, I blindly followed the example you gave me, but that fails in your scenario.

File: demo/cache/cache-controller.js (hunk 1 above)
Comment (kburger) [reply demo/cache/cache-controller.js]: Which line do you mean? 4?

Comment (rajaram5) [reply demo/cache/cache-controller.js]: Yes, line no. 4

**Comment by kburger** [comment]
The only way to solve this without redoing everything was a merge-commit to get my feature branch in sync with develop.

**Review by rajaram5 — APPROVED** [review]

**Pull request merged**